GitCommand/Log.cpp

```diff
@@ -236,49 +225,53 @@ static int add_revision(struct log_state *s, const char *revstr)
 
 
 /**
- * Constructs the entire Log Text for given repo
- * @returns The entire log text to display.
+ * Appends the Log Text for repo to the Log Window
  */
-BString
-Log::GetLogText()
+void
+Log::AppendLogText()
 {
 	git_repository *repo = NULL;
-	BString text;
-	int i, count = 0, printed = 0, parents;
-	struct log_state s;
-	struct log_options opt;
-	git_diff_options diffopts = GIT_DIFF_OPTIONS_INIT;
-	git_oid oid;
-	git_commit *commit = NULL;
 	git_pathspec *ps = NULL;
 
 	git_libgit2_init();
 
-	memset(&s, 0, sizeof(s));
-	s.sorting = GIT_SORT_TIME;
-	s.repodir = fRepo.String();
-
+	struct log_options opt;
+	git_diff_options diffopts = GIT_DIFF_OPTIONS_INIT;
 	memset(&opt, 0, sizeof(opt));
 	opt.max_parents = -1;
 	opt.limit = -1;
 
-	if (!s.revisions)
-		add_revision(&s, NULL);
+	struct log_state state;
+	memset(&state, 0, sizeof(state));
+	state.sorting = GIT_SORT_TIME;
+	state.repodir = fRepo.String();
+
+	if (!state.revisions)
+		add_revision(&state, NULL);
 
 	/** Use the revwalker to traverse the history. */
 
-	printed = count = 0;
+	BString text;
+	git_oid oid;
+	git_commit *commit = NULL;
 
-	for (; !git_revwalk_next(&oid, s.walker); git_commit_free(commit)) {
-		check_lg2(git_commit_lookup(&commit, s.repo, &oid),
+	for (int i = -1; !git_revwalk_next(&oid, state.walker); git_commit_free(commit)) {
+		check_lg2(git_commit_lookup(&commit, state.repo, &oid),
 			"Failed to look up commit", NULL);
-		
+
+		// We only append every ~25 commits because the delay is unnoticable,
```

Comment:
I think this could easily use a larger value (100 or maybe even 1000) and it should still show that something is happening fast enough.

Reply:
You're right, it really makes no difference on big repos-- just set it to 100.